### ISSUE TITLE
Compilation error on fresh Cargo.toml that depends on near-sdk-rs 4.1.1

### ISSUE THREAD

**Issue by ailisp** [near/near-sdk-rs#1119]
Compiling a fresh project depends on near-sdk-rs 4.1.1 (without a Cargo.lock), I get this error:
```
error: no matching package named `parity-secp256k1` found
location searched: registry `crates-io`
required by package `near-crypto v0.14.0`
    ... which satisfies dependency `near-crypto = "^0.14"` of package `near-sdk v4.1.1`
    ... which satisfies dependency `near-sdk = "^4.1.1"` of package `devhub-community-factory v0.1.0 (/Users/bo/workspace/neardevhub-contract/community-factory)`
```

Checking parity-secp256k1 we can see it's yanked. This possibly prevent all new projects from compiling.

**Comment by ailisp** [comment]
I can see the master branch has upgrade to use near-crypto 0.17, which doesn't depend on parity-secp256k1, so I think what we need is a new release

**Comment by ailisp** [comment]
I see a lot of folks are hitting this issue as well. Can we make a new near-sdk-4.x release that simply bump near-crypto to 0.17?

**Comment by fadeevab** [comment]
I hit this issue as well. This issue can also happen in GitHub CI/CD when `Cargo.lock` is not committed.

**Comment by fadeevab** [comment]
@ailisp I believe it's a critical issue as it fails builds completely and there is no workaround, isn't it?

**Comment by frol** [comment]
There is a workaround: https://github.com/near/near-sdk-rs/issues/1126#issuecomment-1879494992

I hope we can get 5.0 release out of the door next week

**Comment by fadeevab** [comment]
@frol A minor update for 4.1.* with fixed dependencies should be valuable as well, as merging to 5.0 involves some breaking changes (so far) which is hard to handle quickly.

**Comment by frol** [comment]
@fadeevab It is not possible to update 4.1.1 as the dependencies are all out of date and there is no breaking-changes-free option there.

**Comment by shreyassai123** [comment]
It can be temporarily fixed until v5 is stable. Add this to `Cargo.toml`
```
[patch.crates-io]
parity-secp256k1 = { git = "https://github.com/paritytech/rust-secp256k1/", version = "0.7.0" }
```

**Comment by frol** [comment]
This is resolved with 5.0 release.
